src/libclipper/include/clipper/logging.hpp

```diff
@@ -0,0 +1,172 @@
+#ifndef CLIPPER_LOGGING_HPP
+#define CLIPPER_LOGGING_HPP
+
+#include <string>
+#include <sstream>
+
+#include <spdlog/spdlog.h>
+
+namespace clipper {
+
+static constexpr uint LOGGING_FORMAT_LENGTH = 18;
+static constexpr uint MAX_TAG_LENGTH = 10;
+// Defines the logging format as [HH:MM:SS.mmm][<LOG_LEVEL_1_CHAR>] <Message>
+// Note: <Message> includes a formatted, user-defined tag (see get_formatted_tag())
+static const std::string LOGGING_FORMAT = "[%T.%e][%L] %v";
+static const std::string LOGGER_NAME = "clipper";
+
+class Logger {
+ public:
+  /**
+   * Constructs a logger that will write all output
+   * to the specified output stream. To be used for
+   * testing!
+   */
+  explicit Logger(std::ostringstream& output_stream);
+  /**
+   * Obtains an instance of the Logger singleton
+   * that can be used to log messages at specified levels
+   */
+  static Logger &get();
+
+  /**
+   * Logs one or more string messages at the "info" log level with the specified tag
+   */
+  template<class ...Strings>
+  void log_info(const std::string tag, Strings... messages) const;
+  /**
+   * Logs a single formatted message at the "info" log level
+   * with the specified tag
+   *
+   * @param args The formatting arguments
+   */
+  template<class ...Args>
```

Comment:
A formatted message takes format arguments. For example, `"The number {}", 5` is a formatted message. The result of the call
`log_info_formatted("TAG", "The number {}", 5)` 
is a log containing the message `"The number 5"`.

 Is there a better word for this?